src/terrain.hpp

```diff
@@ -233,8 +205,8 @@ class Terrain {
 
     void stamp_tile_region(Tile_Stamp tStamp, int x ,int y);
     void stamp_tile_region(int x_start, int y_start, int z_start, int x_end, int y_end, int z_end, const Material * mat, uint8_t color_id);
-    //void stamp_tile_region(int x_start, int y_start, int z_start, int x_end, int y_end, int z_end, const Material * mat, std::set<int> elements_canstamp);
-    void stamp_tile_region(int x_start, int y_start, int z_start, int x_end, int y_end, int z_end, const Material * mat, std::set<std::pair<int, int>> elements_canstamp, uint8_t color_id);
+    //void stamp_tile_region(int x_start, int y_start, int z_start, int x_end, int y_end, int z_end, const Material * mat, std::set<int> elements_can_stamp);
+    void stamp_tile_region(int x_start, int y_start, int z_start, int x_end, int y_end, int z_end, const Material * mat, std::set<std::pair<int, int>> elements_can_stamp, uint8_t color_id);
```

Comment:
Why is this function so long?

Did you know that functions with more than 7 arguments are slower to call?

Reply:
1) it's two vectors that are not packed
2) citation needed
3) if this function is slow it's not the number of arguments (it will be easy to parallelized this)

Reply:
> 1. it's two vectors that are not packed
Oh ok

>     2. citation needed
Look up `x86 calling conventions`

First 6 args are passed through registers, rest through stack. Stack is RAM, even if in L1 cache can take up to 10 cycles to access. 2 of the stack args are pointers, meaning access takes 2 trips to RAM -> lots of cycles wasted.

>     3. if this function is slow it's not the number of arguments (it will be easy to parallelized this)
Slower *to call*.